Formula/k3d.rb

```diff
@@ -0,0 +1,17 @@
+class K3d < Formula
+  desc "Little helper to run Rancher Lab's k3s in Docker"
+  homepage "https://github.com/rancher/k3d"
+  url "https://github.com/rancher/k3d/archive/v1.2.1.tar.gz"
+  sha256 "eb7510aab675183b7febbad79ee23e04de6e6e47369c898ccf03ec2022045709"
+
+  depends_on "go" => :build
+
+  def install
+    system "make", "install-tools"
+    system "make", "BINDIR='#{bin}'", "build"
+  end
+
+  test do
+    system "#{bin}/k3d", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@SMillerDev I've added an additional assertion. All commands will fail if docker is not running.